### PR TITLE
[Refactor] Update custom datasets doc

### PR DESCRIPTION
**Pull request by ZCMax**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Please describe the motivation of this PR and the goal you want to achieve through this PR.

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the back-compatibility of the downstream repos?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
3. If the modification has potential influence on downstream projects, this PR should be tested with downstream projects.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

File: docs/en/advanced_guides/customize_dataset.md [hunk 1]
```diff
@@ -1,367 +1,481 @@
...
-For example, assume the `classes.txt` contains the name of classes as the following.
+#### Visualize your dataset (optional)
 
-```
-person
-bicycle
-car
-```
+To valiate whether your prepared data and config are correct, it's highly recommended to use `tools/browse_dataest.py` script
+to visualize your dataset and annotations before training and validation, more details refer to the visualization doc.
```
Comment (JingweiZhang12) [inline docs/en/advanced_guides/customize_dataset.md:468]: Please add a link of the visualization doc

File: docs/en/advanced_guides/customize_dataset.md [hunk 2]
```diff
@@ -1,367 +1,481 @@
...
+## Evaluation
 
-```python
-classes = 'path/to/classes.txt'
-data = dict(
-    train=dict(classes=classes),
-    val=dict(classes=classes),
-    test=dict(classes=classes))
-```
+Once the data and config has been prepared well, you can directly run the training / testing script following our doc.
```
Comment (JingweiZhang12) [inline docs/en/advanced_guides/customize_dataset.md:472]:
```suggestion
Once the data and config have been prepared, you can directly run the training/testing script following our doc.
```

File: docs/en/advanced_guides/customize_dataset.md [hunk 3]
```diff
@@ -1,367 +1,481 @@
...
 
-**Note** (related to MMDetection):
+**Note**: we only provide an implementation for KITTI stype evaluation for customized dataset. It should be included in dataset config:
```
Comment (JingweiZhang12) [inline docs/en/advanced_guides/customize_dataset.md:474]:
```suggestion
**Note**: we only provide an implementation for KITTI style evaluation for the customized dataset. It should be included in the dataset config:
```

File: docs/en/advanced_guides/customize_dataset.md [hunk 4]
```diff
@@ -1,367 +1,481 @@
...
 
-**Note:**
+Regarding the setting of `anchor_range`, it is generally adjusted according to dataset. Note that `z` value needs to be adjusted accordingly to the position of the point cloud, please refer to this [issue](https://github.com/open-mmlab/mmdetection3d/issues/986).
 
-1. The option `separate_eval=False` assumes the datasets use `self.data_infos` during evaluation. Therefore, COCO datasets do not support this behavior since COCO datasets do not fully rely on `self.data_infos` for evaluation. Combining different types of datasets and evaluating them as a whole is not tested thus is not suggested.
-2. Evaluating `ClassBalancedDataset` and `RepeatDataset` is not supported thus evaluating concatenated datasets of these types is also not supported.
+Regarding the setting of `anchor_size`, it is usually necessary to count the average length, width and height of the entire training dataset as `anchor_size` to obtain the best results.
```
Comment (JingweiZhang12) [inline docs/en/advanced_guides/customize_dataset.md:345]:
```suggestion
Regarding the setting of `anchor_size`, it is usually necessary to count the average length, width and height of objects in the entire training dataset as `anchor_size` to obtain the best results.
```

File: docs/en/advanced_guides/customize_dataset.md [hunk 5]
```diff
@@ -1,367 +1,481 @@
...
-       'Tr_imu_to_velo': array([[ 9.999976e-01,  7.553071e-04, -2.035826e-03, -8.086759e-01],
-       [-7.854027e-04,  9.998898e-01, -1.482298e-02,  3.195559e-01],
-       [ 2.024406e-03,  1.482454e-02,  9.998881e-01, -7.997231e-01],
-       [ 0.000000e+00,  0.000000e+00,  0.000000e+00,  1.000000e+00]])},
-     'annos': {'name': array(['Pedestrian'], dtype='<U10'), 'truncated': array([0.]), 'occluded': array([0]), 'alpha': array([-0.2]), 'bbox': array([[712.4 , 143.  , 810.73, 307.92]]), 'dimensions': array([[1.2 , 1.89, 0.48]]), 'location': array([[1.84, 1.47, 8.41]]), 'rotation_y': array([0.01]), 'score': array([0.]), 'index': array([0], dtype=int32), 'group_ids': array([0], dtype=int32), 'difficulty': array([0], dtype=int32), 'num_points_in_gt': array([377], dtype=int32)}}
-    ...
-]
+#### Label Format
+
+The most basic information: 3D bounding box and category label of each scene need to be contained in annotation `.txt` file. Each line represents a 3D box in a given scene as follow:
```
[the remark this answers is not in the document]
Comment (ZCMax) [reply docs/en/advanced_guides/customize_dataset.md]: I think `in each scene` is not a proper description, I change it to `a certain scene`

**Review by JingweiZhang12 — APPROVED** [review]

File: docs/en/advanced_guides/customize_dataset.md [hunk 6]
```diff
@@ -1,367 +1,482 @@
...
-       'Tr_imu_to_velo': array([[ 9.999976e-01,  7.553071e-04, -2.035826e-03, -8.086759e-01],
-       [-7.854027e-04,  9.998898e-01, -1.482298e-02,  3.195559e-01],
-       [ 2.024406e-03,  1.482454e-02,  9.998881e-01, -7.997231e-01],
-       [ 0.000000e+00,  0.000000e+00,  0.000000e+00,  1.000000e+00]])},
-     'annos': {'name': array(['Pedestrian'], dtype='<U10'), 'truncated': array([0.]), 'occluded': array([0]), 'alpha': array([-0.2]), 'bbox': array([[712.4 , 143.  , 810.73, 307.92]]), 'dimensions': array([[1.2 , 1.89, 0.48]]), 'location': array([[1.84, 1.47, 8.41]]), 'rotation_y': array([0.01]), 'score': array([0.]), 'index': array([0], dtype=int32), 'group_ids': array([0], dtype=int32), 'difficulty': array([0], dtype=int32), 'num_points_in_gt': array([377], dtype=int32)}}
-    ...
-]
+#### Label Format
+
+The most basic information: 3D bounding box and category label of each scene need to be contained in annotation `.txt` file. Each line represents a 3D box in a certain scene as follow:
```
Comment (Tai-Wang) [inline docs/en/advanced_guides/customize_dataset.md:26]: in the annotation `.txt` file

File: docs/en/advanced_guides/customize_dataset.md [hunk 7]
```diff
@@ -1,367 +1,482 @@
...
-```python
-import numpy as np
-from os import path as osp
+#### LiDAR-Based 3D Detection
 
-from mmdet3d.core import show_result
-from mmdet3d.core.bbox import DepthInstance3DBoxes
-from mmdet.datasets import DATASETS
-from .custom_3d import Custom3DDataset
+The raw data for LiDAR-Based 3D object detection are typically organized as follows, where `ImageSets` contains split files indicating which files belong to training/validation set, `points` include point cloud data which are supposed to be stored in `.bin` format and `labels` includes label files for 3D detection.
```
Comment (Tai-Wang) [inline docs/en/advanced_guides/customize_dataset.md:62]: is typically organized as follows

Comment (Tai-Wang) [reply docs/en/advanced_guides/customize_dataset.md:62]: `points` include -> includes

Comment (Tai-Wang) [reply docs/en/advanced_guides/customize_dataset.md:62]: includes the `.bin` point cloud data

File: docs/en/advanced_guides/customize_dataset.md [hunk 8]
```diff
@@ -1,367 +1,482 @@
...
-
-        anns_results = dict(
-            gt_bboxes_3d=gt_bboxes_3d,
-            gt_labels_3d=gt_labels_3d,
-            pts_instance_mask_path=pts_instance_mask_path,
-            pts_semantic_mask_path=pts_semantic_mask_path)
-        return anns_results
+#### Vision-Based 3D Detection
+
+The raw data for Vision-Based 3D object detection are typically organized as follows, where `ImageSets` contains split files indicating which files belong to training/validation set, `images` contains the images from different cameras, for example, images from `camera_x` need to be placed in `images\images_x`. `calibs` contains calibration information files which store the camera intrinsic matrix of each camera, and `labels` includes label files for 3D detection.
```
Comment (Tai-Wang) [inline docs/en/advanced_guides/customize_dataset.md:86]: vision-based 3D object detection
(no need to use block letters)

Comment (Tai-Wang) [reply docs/en/advanced_guides/customize_dataset.md:86]: is typically organized

File: docs/en/advanced_guides/customize_dataset.md [hunk 9]
```diff
@@ -1,367 +1,482 @@
...
+│   │   ├── labels
+│   │   │   ├── 000000.txt
+│   │   │   ├── 000001.txt
+│   │   │   ├── ...
 ```
 
-Then in the config, to use `MyDataset` you can modify the config as the following
+#### Multi-Modality 3D Detection
+
+The raw data for Multi-Modality 3D object detection are typically organized as follows. Different from Vision-based 3D Object detection, calibration information files in `calibs` store the camera intrinsic matrix of each camera and extrinsic matrix.
```
Comment (Tai-Wang) [inline docs/en/advanced_guides/customize_dataset.md:118]: is typically

Comment (Tai-Wang) [inline docs/en/advanced_guides/customize_dataset.md:118]: Multi-Modality -> multi-modality
Vision -> vision

**Review by Tai-Wang — APPROVED** [review]

**Pull request merged**